src/gui/FileBrowser.cpp

```diff
@@ -229,22 +229,21 @@ void FileBrowser::reloadTree()
 
 
 
-void FileBrowser::expandItems( QTreeWidgetItem * item, QList<QString> expandedDirs )
+void FileBrowser::expandItems(QTreeWidgetItem* item, QList<QString> expandedDirs)
 {
 	int numChildren = item ? item->childCount() : m_fileBrowserTreeWidget->topLevelItemCount();
 	for (int i = 0; i < numChildren; ++i)
 	{
-		QTreeWidgetItem * it = item ? item->child( i ) : m_fileBrowserTreeWidget->topLevelItem(i);
+		auto it = item ? item->child(i) : m_fileBrowserTreeWidget->topLevelItem(i);
 		auto d = dynamic_cast<Directory*>(it);
 		if (d)
 		{
-			d->update();
-			bool expand = expandedDirs.contains( d->fullName() );
-			d->setExpanded( expand );
-		}
-		if (m_recurse && it->childCount())
-		{
-			expandItems(it, expandedDirs);
+			if (m_recurse) { d->setExpanded(true); }
+			d->setExpanded(expandedDirs.contains(d->fullName()));
```

Comment:
I just realized this is doing the same thing, and it's possible that `m_recurse` is ignored in the case `expandedDirs.contains(d->fullName())` returns `false`.
Is this what we want to do here then?
```suggestion
			d->setExpanded(m_recurse || expandedDirs.contains(d->fullName()));
```

Reply:
Confusion over this exact area is actually exactly what caused the previous PR to break things.  Using "setExpanded(true)" and then "setExpanded(false)" immediately after actually does have an impact contrary to what you'd expect, since doing so goes through and caches all the files within that directory.  The change proposed here causes all directories to be expanded by default, since they aren't collapsed after being expanded anymore.

Reply:
Gotcha! All good then.